src/artm/regularizer_sandbox/decorrelator_phi.cc

```diff
@@ -17,31 +18,11 @@ bool DecorrelatorPhi::RegularizePhi(::artm::core::Regularizable* topic_model, do
   // read the parameters from config and control their correctness
   const int topic_size = topic_model->topic_size();
 
-  auto topic_name = topic_model->topic_name();
-  std::vector<bool> topics_to_regularize;
-  if (config_.topic_name_size() > 0) {
-    for (int i = 0; i < topic_size; ++i)
-      topics_to_regularize.push_back(false);
-
-    for (int topic_id = 0; topic_id < config_.topic_name_size(); ++topic_id) {
-      for (int real_topic_id = 0; real_topic_id < topic_size; ++real_topic_id) {
-        if (topic_name.Get(real_topic_id) == config_.topic_name(topic_id)) {
-          topics_to_regularize[real_topic_id] = true;
-          break;
-        }
-      }
-    }
-  } else {
-    for (int i = 0; i < topic_size; ++i)
-      topics_to_regularize.push_back(true);
-  }
+  std::vector<bool> topics_to_regularize = core::is_member(config_.topic_name(),
```

Comment:
Or just introduce "use_all_topics" flag and check "if (use_all_topics || topics_to_regularize[topic_id])". This is safe because C++ guaranties to NOT EVALUATE second part of "if (A || B)" whenever A evaluates to true.

http://stackoverflow.com/questions/1331318/how-an-if-a-b-statement-is-evaluated
In C and C++, the && and || operators "short-circuit". That means that they only evaluate a parameter if required. If the first parameter to && is false, or the first to || is true, the rest will not be evaluated.